### PR TITLE
#Task1_4 Yury.Lagodich

### PR DESCRIPTION
**Pull request by zym43210**
https://travis-ci.com/github/zym43210/js-assignments/builds/174341910

### CODE REVIEW

**Comment by AlexMokin** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by zym43210** [comment]
I cant understand what is wrong, all tests are passed from 1 to 4 tasks

**Comment by AlexMokin** [comment]
Please correct code according to the comments

**Comment by zym43210** [comment]
i didnt get any comments, so i tried to correct my code

**Comment by AlexMokin** [comment]
1) in strings task don't use regular expressions, there is a simple solution without regular expressiosn for all these tasks.
2) See documentation for the Math.hypot function https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/hypot. Use it in methods
3) there are lots of missing ;
4) rewrite getPositivesCount with Array.reduce https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce
5) remove usages of outdated "var" keyword. Replace with either "let" or "const"
6) replace all code similar to this 

`if (<bolean expression>) {
        return true;
    }
    else {
        return false;
    }`

with 

`return <bolean expression>`

**Comment by zym43210** [comment]
I am sorry but i didnt find 5-th point in my solution

**Comment by AlexMokin** [comment]
image
looks good, just fix the indentation here. You shouldn't change format of the existing code

**Comment by zym43210** [comment]
I am so sorry but i cant inderstand what's wrong with indention or where should i fix this and how.

**Comment by zym43210** [comment]
image

if u mean this one, so i didn't touch this

**Comment by AlexMokin** [comment]
yes it's these line, they are changed, see this diff https://github.com/AisBrestEDU/js-assignments/pull/228/files#diff-0f7e89351c11cd1290aec469b3f72b7b

Just copy them over form the original file

**Comment by zym43210** [comment]
I am not sure but looks like fixed this

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
 function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+    return value.substring(7, value.length - 1);
```
Comment (FedorovD) [inline task/01-strings-tasks.js:72]: looks good, but please note substring can work with negative numbers also to take symbols from end. In the case `substring(7, -1);` can do the same.

File: task/04-arrays-tasks.js [hunk 2]
```diff
@@ -249,7 +289,13 @@ function getMovingSum(arr) {
  * [ "a" ] => []
  */
 function getSecondItems(arr) {
-   throw new Error('Not implemented');
+   const resArr = [];
+   arr.map((elem, index) => {
+      if (index % 2 != 0) {
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:294]: operator `===` / `!==`is always preferable

**Comment by FedorovD** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**